### ISSUE TITLE
`handle_proposal` allows spamming root block to evict legitimate proposals from buffer

### ISSUE THREAD

**Issue by technovision99** [category-labs/monad-bft#1334]
When receiving a proposal request, the `verify_and_validate_consensus_message` function will be called which verifies the signature, epoch, round, and certificates on the block. Importantly it does not verify if the block has already been seen or is stored in the blocktree. The `handle_proposal` function is then called: 

```rust
pub fn handle_proposal(
        &mut self,
        author: NodeId<SCT::NodeIdPubKey>,
        proposal: ProposalMessage<SCT>,
    ) -> Option<(RootInfo, QuorumCertificate<SCT>)> {
        // TODO more validation? leader checking? more sophisticated eviction?
        if proposal.block.get_seq_num() < self.root {
            return None;
        }
        let block_header = proposal.block.clone();
        self.block_headers
            .insert(proposal.block.get_id(), block_header.clone());
        self.proposal_buffer.push_back((author, proposal));
        if self.proposal_buffer.len() > self.max_buffered_proposals {
            self.proposal_buffer.pop_front();
        }
 ```

Notably the function does not reject any duplicate block headers. As a result a malicious validator can spam proposals to evict legitimate proposals from the `proposal_buffer` and only process sync requests for the same root over and over again. 
  Additionally I think that the function should actually use `<=` when checking `proposal.block.get_seq_num() < self.root` because `self.root` should already be committed.

**Comment by omegablitz** [comment]
Yes this is correct, we still need to harden the statesync-related code. monad-state/src/statesync.rs is quite primitive right now, we will work to improve this.

**Comment by maxkozlovsky** [comment]
It seems that loosing entries from proposal_buffer is not that big deal, this is used as cache anyway. The field is not used to determine the statesync target, which makes it not possible to force a wrong statesync target. It does not look worth it to look for better replacement algorithm since the code has to deal with proposals not present in proposal_buffer anyway.

**Comment by maxkozlovsky** [comment]
Suggest closing or postponing until we have enough resources to look at better eviction algorithm.

**Comment by sestruck** [comment]
Close confirmed with maxkozlovsky 9/16
